cfpb/consumerfinance.gov
# Add a max length for Ask search fields

**This is a work in progress**. It still needs unit and functional tests, but before that, I'm hoping a front-end developer can give everything a look to see if the code is on the right track and suggest any improvements.

Adds a max length to Ask search fields to prevent visitors from accidentally trying to enter complaints there. See GHE/CFGOV/platform/issues/4091 for more details.

---

## Additions

- Max length to the autocomplete molecule and Ask search organism

## How to test this PR

1. Go to the complaint landing page, `/complaint/`. Edit that page in Wagtail and check the "Direct long searches to submit a complaint" checkbox. Publish that change.
2. Try pasting or typing more than 50 characters into the Ask search field. You should see a field-level error and be prevented from entering any more text, like in the screenshot below.
3. Once you have 50 characters in the box, try deleting one or two. You should get autocomplete results as normal again.
4. Try deleting all but one character. Autocomplete results should disappear until you type a second character (as they do now).
5. Go to another page with an autocomplete-enabled Ask search field, like `/ask-cfpb/`. Try pasting or typing more than 50 characters into that field. You should get the same error, but the error message should only show the first sentence ("Searches are limited to 50 characters.").

## Screenshots

![search-error](https://user-images.githubusercontent.com/1862695/127216343-5f315a5f-6152-418c-915b-e13eb2e010c3.gif)

## Notes and todos

- [x] Get answers to the slew of questions I put in the code as TODO comments
- [x] Check all the browser testing steps down below
- [x] ~~Write unit tests~~ (there aren't any existing unit tests for any of the files I changed, so I'm going to keep that task as an effort separate from this PR
- [x] Write functional tests
- [x] Set up a custom analytics event
- [x] Replicate a similar max length validation on the back end to make this work without Javascript. I'll put that in a separate PR, though.

## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing

#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
